### PR TITLE
Testcloud plugin: Make ProvisionError more verbose

### PR DESCRIPTION
**Pull request by ZhukovGreen**
ref #634 

Increase the verbosity of the exceptions

### CODE REVIEW

**Comment by ZhukovGreen** [comment]
Probably will use 'raise from'. It will be better. I'll test it on Monday

**Comment by ZhukovGreen** [comment]
Exception chaining is not making the exception more readable in our particular case. It seems it is due to the way the exception is reported. We can leave this MR as it is then

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -397,12 +397,15 @@ def start(self):
...
+        except FileNotFoundError as e:
+            raise ProvisionError(
+                f"Image '{self.image.local_path}' not found.\n {str(e)}"
+            )
+        except testcloud.exceptions.TestcloudPermissionsError as e:
             raise ProvisionError(
                 f"Failed to prepare the image. "
-                f"Check the '{TESTCLOUD_IMAGES}' directory permissions.")
+                f"Check the '{TESTCLOUD_IMAGES}' directory permissions.\n"
+                f"{str(e)}")
```
Comment (lukaszachy) [inline tmt/steps/provision/testcloud.py:408]: Currently it prints 
```
Image '/var/tmp/tmt/testcloud/images/a.qcow2' not found.
 [Errno 2] No such file or directory: 'selinuxenabled'
```

Personally I'd prefix something so it prints e.g.
```
Image '/var/tmp/tmt/testcloud/images/a.qcow2' not found.
Raised exception was:  [Errno 2] No such file or directory: 'selinuxenabled'
```

@psss thoughts?

Comment (psss) [reply tmt/steps/provision/testcloud.py:408]: Yes, something like that will be better. I think this is quite common scenario. Let's make handling of such exceptions a bit more comfortable. See 2d33f7e.

**Review by psss — APPROVED** [review]
Thanks for improving this. As showing the original exception details is quite common I've added a couple of adjustments on top of your commit. @ZhukovGreen, @lukaszachy, please have a look.

**Review by lukaszachy — APPROVED** [review]
LGTM

**Comment by ZhukovGreen** [comment]
Thx guys for improving this!

**Pull request merged**